bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/util/ActionInputsHelper.java

```diff
@@ -0,0 +1,327 @@
+                    break;
+                case "java.lang.String":
+                    break;
+                case "java.time.LocalDate":
+                    context = "date";
+                    break;
+                case "java.time.LocalTime":
+                    context = "time";
+                    break;
+                case "java.time.LocalDateTime":
+                case "java.util.Date":
+                    context = "datetime";
+                    break;
+                case "java.time.ZonedDateTime":
+                case "java.time.Instant":
+                case "java.time.Duration":
+                    // There is no available configuration parameter context for these types.
+                    // A text parameter is used. The expected value must respect a particular format specific
+                    // to each of these types.
+                    break;
+                default:
+                    supported = false;
+                    break;
+            }
+        }
+        if (!supported) {
+            throw new IllegalArgumentException("Input parameter '" + input.getName() + "' with type " + input.getType()
+                    + "cannot be converted into a config description parameter!");
+        }
+
+        ConfigDescriptionParameterBuilder builder = ConfigDescriptionParameterBuilder
+                .create(input.getName(), parameterType).withLabel(input.getLabel())
+                .withDescription(input.getDescription()).withReadOnly(false)
+                .withRequired(required || input.isRequired()).withContext(context);
+        if (!input.getDefaultValue().isEmpty()) {
+            builder = builder.withDefault(input.getDefaultValue());
+        } else if (defaultValue != null) {
+            builder = builder.withDefault(defaultValue);
+        }
+        if (unit != null) {
+            builder = builder.withUnit(unit.getSymbol());
+        }
+        return builder.build();
+    }
+
+    /**
+     * Maps serialised inputs to the Java types required by the {@link Input}s of the given {@link ActionType}.
+     *
+     * @param actionType the action type whose inputs to consider
+     * @param arguments the serialised arguments
+     * @return the mapped arguments
+     */
+    public Map<String, Object> mapSerializedInputsToActionInputs(ActionType actionType, Map<String, Object> arguments) {
+        Map<String, Object> newArguments = new HashMap<>();
+        for (Input input : actionType.getInputs()) {
+            String name = input.getName();
+            Object value = arguments.get(name);
+            try {
+                value = mapSerializedInputToActionInput(actionType, input, value);
+            } catch (IllegalArgumentException e) {
+                logger.warn("{} Input parameter is ignored.", e.getMessage());
+                value = null;
+            }
+            if (value == null) {
+                continue;
+            }
+            newArguments.put(name, value);
```

Comment:
```suggestion
            try {
                newArguments.put(name, mapSerializedInputToActionInput(actionType, input, value));
            } catch (IllegalArgumentException e) {
                logger.warn("{} Input parameter is ignored.", e.getMessage());
            }           
```

Since the return value is no longer `@Nullable` you can simplify the code here. Maybe you can also remove the `value` and `input` variables.

Reply:
Done